ducktape/tests/runner.py

```diff
@@ -132,24 +137,30 @@ def run_all_tests(self):
         self._log(logging.INFO, "running %d tests..." % len(self.scheduler))
 
         while self._ready_to_trigger_more_tests or self._expect_client_requests:
-
-            while self._ready_to_trigger_more_tests:
-                next_test_context = self.scheduler.next()
-                self._preallocate_subcluster(next_test_context)
-                self._run_single_test(next_test_context)
-
-            if self._expect_client_requests:
-                try:
-                    event = self.receiver.recv()
-                    self._handle(event)
-                except Exception as e:
-                    err_str = "Exception receiving message: %s: %s" % (str(type(e)), str(e))
-                    err_str += "\n" + traceback.format_exc(limit=16)
-                    self._log(logging.ERROR, err_str)
-                    continue
+            try:
+                while self._ready_to_trigger_more_tests:
+                    next_test_context = self.scheduler.next()
+                    self._preallocate_subcluster(next_test_context)
+                    self._run_single_test(next_test_context)
+
+                if self._expect_client_requests:
+                    try:
+                        event = self.receiver.recv()
+                        self._handle(event)
+                    except Exception as e:
+                        err_str = "Exception receiving message: %s: %s" % (str(type(e)), str(e))
+                        err_str += "\n" + traceback.format_exc(limit=16)
+                        self._log(logging.ERROR, err_str)
+
+                        # All processes are on the same machine, so treat communication failure as a fatal error
+                        raise
+            except KeyboardInterrupt:
+                # If SIGINT is received, stop triggering new tests, and let the currently running tests finish
+                self.stop_testing = True
```

Comment:
you might want to log something here too -- it'll at least let them know that you're working on shutting things down.